### PR TITLE
WIP: more functionalities, repository cleanup

### PR DESCRIPTION
**Pull request by Dzordzu**
### Quota actions
It's not applied in the GO SDK, but it would be nice to enable setting quota programmatically

## Applied updates
- [x] Added `get_bucket_quota`, `set_bucket_quota` #35 
- [x] Simplified testing (detected some issues with tmpfs and minio; added docker based alternative) #32 
- [x] Fixed booleans from the env within the test file #32 
- [x] Upgraded dependencies #32 
- [x] fmt and clippy (reasons behind pr being so large) #32

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -2,3 +2,4 @@
 **/*.rs.bk
 Cargo.lock
 .idea
+minio
```
Comment (Dzordzu) [inline .gitignore:5]: Ignoring binary created during manual tests

**Review by balamurugana — CHANGES_REQUESTED** [review]
Send cleanup, new API addition and other fixes as individual PRs

File: src/s3/client.rs [hunk 2]
```diff
@@ -2355,6 +2361,76 @@ impl<'a> Client<'a> {
...
+
+        let data = serde_json::to_string(&args.quota)?;
+
+        let resp = self
+            .execute(
+                Method::PUT,
+                &"us-east-1".into(),
+                &mut headers,
+                &query_params,
+                "minio/admin/v3/set-bucket-quota".into(),
```
Comment (balamurugana) [inline src/s3/client.rs:2386]: You should keep Minio admn APIs as separate package like [minio-java](https://github.com/minio/minio-java/tree/master/adminapi/src) or [minio-py](https://github.com/minio/minio-py/blob/master/minio/minioadmin.py); not inside `s3`.

Comment (Dzordzu) [reply src/s3/client.rs:2386]: Maybe just move it to the separate module?

Comment (balamurugana) [reply src/s3/client.rs:2386]: like `src/admin` or `src/adminapi`

Comment (Dzordzu) [reply src/s3/client.rs:2386]: Yep. That's what I mean

**Comment by Dzordzu** [comment]
Okey. With the incoming commit everything seems green. Check it locally. I've been working for 10h. My weekend looked almost the same. Sorry again for this mess
image

**Pull request closed without merging**